### PR TITLE
Fix files shown as IDLE when opened with CustomEditor, add support for *.drawio files

### PR DESCRIPTION
**Pull request by cathaypacific8747**
When opening [*.drawio](https://github.com/hediet/vscode-drawio) files, this extension incorrectly shows the discord status as "Idling":

idle

This also occurs when you preview or edit any file in a [non-text-based editor](https://code.visualstudio.com/api/extension-guides/webview), affecting wide range of files (such as PNG, JPG etc.)

After debugging, it appears to me that `window.activeTextEditor` will always cast to `undefined` [when being opened in such an editor](https://stackoverflow.com/a/65114438), causing many parts of the code to be unable to retrieve the file name, file location and other information. This PR fixes this issue by [obtaining the selected file's URI indirectly](https://github.com/microsoft/vscode/issues/3553#issuecomment-1098562676), then parsing the required information. I've also added the *.drawio icons in this PR.

fixed

Hopefully this will be helpful :)

### CODE REVIEW

File: src/data/languages.json [hunk 1]
```diff
@@ -171,6 +171,7 @@
 		".dart": { "image": "dart" },
 		".dfm": { "image": "delphi" },
 		".dpr": { "image": "delphi" },
+		"/\\.(drawio|dio)(\\.png|\\.svg)?/i": { "image": "drawio" },
```
Comment (cathaypacific8747) [inline src/data/languages.json:174]: [Support](https://github.com/hediet/vscode-drawio/blob/8da9fb55cc55825fa4d2778f79b7f534a8d77bc8/src/DrawioEditorService.ts#L170-L177) `*.drawio`, `*.dio`,`*.drawio.png`, `*.dio.svg`, `*.drawio.png`,`*.dio.svg` files

File: src/activity.ts [hunk 2]
```diff
@@ -50,7 +44,18 @@ export function onDiagnosticsChange() {
...
+async function getFileURI() {
+	// temporary workaround to get the file URI of the current active CustomTextEditor
+	// https://github.com/microsoft/vscode/issues/3553#issuecomment-1098562676
+
+	const origClipboard = await env.clipboard.readText();
+	await commands.executeCommand('copyFilePath');
+	const filePath = await env.clipboard.readText();
+	await env.clipboard.writeText(origClipboard);
+	return Uri.file(filePath);
+}
```
Comment (cathaypacific8747) [inline src/activity.ts:56]: Indirectly obtains the file's URI, which would then be used for resolving the file's icon if the `window.activeTextEditor` object is `undefined`.

File: src/activity.ts [hunk 3]
```diff
@@ -277,27 +264,32 @@ function details(
...
-		);
+	} else if (_fileIcon !== undefined) {
+		// since this file is now edited with a CustomTextEditor, the following limitations apply:
+		// - unable to distinguish between debugging, viewing or editing -> editing
+		// - unable to show TotalLines, CurrentLine and CurrentColumn -> zeroed / emptied
+
+		raw = config[editing]
+			.replace(REPLACE_KEYS.TotalLines, FAKE_EMPTY)
+			.replace(REPLACE_KEYS.CurrentLine, '0')
+			.replace(REPLACE_KEYS.CurrentColumn, '0')
```
Comment (cathaypacific8747) [inline src/activity.ts:275]: In the case that the file is opened with a `CustomEditor`, it is impossible to obtain the current line number / column etc., so populate it with zero.

**Comment by LeonardSSH** [comment]
Thanks for this PR, everything seems to be working properly. Just a small idling problem, but I fixed it in the last commit. 👍

**Review by LeonardSSH — APPROVED** [review]

**Pull request merged**